ray_on_golem/client/client.py

```diff
@@ -19,6 +25,9 @@ def __init__(self, base_url: URL) -> None:
 
         self._session = requests.Session()
 
+        if not is_running_on_golem_network():
+            self._start_webserver()
```

Comment:
on a separate note - question - what if the webserver startup fails when it's required? wouldn't we want to raise an explicit exception?